compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/GetTextFieldPointerModifier.uikit.kt

```diff
@@ -0,0 +1,137 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text
+
+import androidx.compose.foundation.gestures.detectDragGesturesAfterLongPress
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.text.selection.SelectionAdjustment
+import androidx.compose.foundation.text.selection.TextFieldSelectionManager
+import androidx.compose.foundation.text.selection.getTextFieldSelection
+import androidx.compose.foundation.text.selection.isSelectionHandleInVisibleBound
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.hapticfeedback.HapticFeedbackType
+import androidx.compose.ui.input.pointer.pointerHoverIcon
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TextFieldValue
+
+internal actual fun getTextFieldPointerModifier(
+    manager: TextFieldSelectionManager,
+    enabled: Boolean,
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping
+): Modifier = if (enabled) {
+    if (isInTouchMode) {
+        val selectionModifier = getSelectionModifier(manager)
+        val tapHandlerModifier = getTapHandlerModifier(
+            interactionSource,
+            state,
+            focusRequester,
+            readOnly,
+            offsetMapping,
+            manager
+        )
+        Modifier
+            .then(tapHandlerModifier)
+            .then(selectionModifier)
+            .pointerHoverIcon(textPointerIcon)
+    } else {
+        Modifier
+            .mouseDragGestureDetector(
+                observer = manager.mouseSelectionObserver,
+                enabled = enabled
+            )
+            .pointerHoverIcon(textPointerIcon)
+    }
+} else {
+    Modifier
+}
+
+@OptIn(InternalFoundationTextApi::class)
+private fun getTapHandlerModifier(
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping,
+    manager: TextFieldSelectionManager
+) = Modifier.then(Modifier.pointerInput(interactionSource) {
```

Comment:
It looks like `then` is not required here

Reply:
@mazunin-v-jb Yes, we can simplify it